desafio-06/disouzam/python/anagrama.py

```diff
@@ -0,0 +1,469 @@
+"""
+    Processa uma palavra ou frase e retorna todas as combinações possíveis de anagramas com palavras
+    presentes no arquivo desafio-06/disouzam/python/words.txt (cópia local do arquivo disponível em
+    https://osprogramadores.com/desafios/d06/words.txt)
+"""
+from ctypes import ArgumentError
+import os
+import string
+import sys
+import re
+
+
+def main(args):
+    """main(args):
+    Processa uma expressão passada na linha de comando, descrita pelo parâmetro args
+    e imprime os anagramas existentes a partir de uma busca no arquivo words.txt.
+
+    Parâmetros:
+    args: Lista de argumentos recebido da linha de comando e
+          pré-processado na chamada da função main. Deve conter 2 argumentos e
+          o segundo é uma expressão a ser processada em busca dos anagramas.
+    """
+    expressao = ""
+
+    # Análise dos argumentos recebidos em args
+    if len(args) <= 1:
+        raise ArgumentError("Nenhum argumento foi fornecido.")
+
+    # Validação dos argumentos
+    if len(args) >= 2:
+        expressao = args[1]
+        expressao = expressao.upper().replace(" ", "")
+        if not expressao_e_valida(expressao):
+            raise ArgumentError("Expressão contém caracteres inválidos")
+
+    if len(args) > 2:
+        mensagem1 = f"Você informou um número excessivo de argumentos ({
+            len(args) - 1}). "
```

Comment:
Esta quebra de linha aqui está quebrando a execução do programa. Creio que foi sem querer.

Vou usar essa situação para lhe explicar sobre a funcionalidade do review:

1. Vou solicitar mudanças.
2. Você faz o ajuste e envia o commit
3. Após isso, você mesmo já vai no campo de reviews e clica no botão que tem as setinhas. Ele vai solicitar um novo review para aquela pessoa.

Reply:
O botão que tem as setinhas é este aqui:
![image](https://private-user-images.githubusercontent.com/4745542/320214963-d7e28e5e-d449-4adc-adf1-4f0ad9418485.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI0MTc3OTksIm5iZiI6MTcxMjQxNzQ5OSwicGF0aCI6Ii80NzQ1NTQyLzMyMDIxNDk2My1kN2UyOGU1ZS1kNDQ5LTRhZGMtYWRmMS00ZjBhZDk0MTg0ODUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDZUMTUzMTM5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDFjOTI0MDU5NDdlYWYxMjJhODA2NzM0NmZiNzU3ZjVlODIzN2IxMzliYTkxMGI1MzFjY2VjMjlmZGNkYTE0NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.fs5EysTiYmc-Mt1BuQhuBvlTy3o3KjYMBaDylRV1OkM)

Reply:
@magnussolidus 

Qual versão do Python você está usando? Se não for a versão 3.12, sim, vai quebrar!

Reply:
A mais recente disponível em minha distro é a 3.11.8.

De qualquer forma é algo simples. O mais importante aqui é a questão de solicitar o review. 

Eu creio que você conseguiu fazer, então vou marcar aqui como resolvido.

Reply:
Então é importante revisar o texto do desafio. E meu README deixa explícito que foi testado na versão 3.12.1.